pkg/scheduler/framework/framework_test.go

```diff
@@ -0,0 +1,297 @@
+			Name: "binding-7",
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			State:                        fleetv1beta1.BindingStateBound,
+			TargetCluster:                clusterName1,
+			SchedulingPolicySnapshotName: policyName,
+		},
+	}
+	scheduledBinding := fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "binding-8",
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			State:                        fleetv1beta1.BindingStateScheduled,
+			TargetCluster:                clusterName2,
+			SchedulingPolicySnapshotName: policyName,
+		},
+	}
+
+	bindings := []fleetv1beta1.ClusterResourceBinding{
+		unscheduledBinding,
+		associatedWithLeavingClusterBinding,
+		assocaitedWithDisappearedClusterBinding,
+		obsoleteBinding,
+		boundBinding,
+		scheduledBinding,
+	}
+	wantBound := []*fleetv1beta1.ClusterResourceBinding{&boundBinding}
+	wantScheduled := []*fleetv1beta1.ClusterResourceBinding{&scheduledBinding}
+	wantObsolete := []*fleetv1beta1.ClusterResourceBinding{&obsoleteBinding}
+	wantDangling := []*fleetv1beta1.ClusterResourceBinding{&associatedWithLeavingClusterBinding, &assocaitedWithDisappearedClusterBinding}
+
+	bound, scheduled, obsolete, dangling := classifyBindings(policy, bindings, clusters)
+	if diff := cmp.Diff(bound, wantBound); diff != "" {
+		t.Errorf("classifyBindings() bound diff (-got, +want): %s", diff)
+	}
+
+	if diff := cmp.Diff(scheduled, wantScheduled); diff != "" {
+		t.Errorf("classifyBindings() scheduled diff (-got, +want) = %s", diff)
+	}
+
+	if diff := cmp.Diff(obsolete, wantObsolete); diff != "" {
+		t.Errorf("classifyBindings() obsolete diff (-got, +want) = %s", diff)
+	}
+
+	if diff := cmp.Diff(dangling, wantDangling); diff != "" {
+		t.Errorf("classifyBindings() dangling diff (-got, +want) = %s", diff)
+	}
+}
+
+// TestMarkAsUnscheduled tests the markAsUnscheduled method.
+func TestMarkAsUnscheduled(t *testing.T) {
+	binding := fleetv1beta1.ClusterResourceBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: bindingName,
+		},
+		Spec: fleetv1beta1.ResourceBindingSpec{
+			State: fleetv1beta1.BindingStateBound,
+		},
+	}
+
+	fakeClient := fake.NewClientBuilder().
+		WithScheme(scheme.Scheme).
+		WithObjects(&binding).
+		Build()
+	// Construct framework manually instead of using NewFramework() to avoid mocking the controller manager.
+	f := &framework{
+		client: fakeClient,
+	}
+
+	ctx := context.Background()
+	if err := f.markAsUnscheduledFor(ctx, []*fleetv1beta1.ClusterResourceBinding{&binding}); err != nil {
+		t.Fatalf("markAsUnscheduledFor() = %v, want no error", err)
+	}
+
+	if err := fakeClient.Get(ctx, types.NamespacedName{Name: bindingName}, &binding); err != nil {
+		t.Fatalf("Get cluster resource binding %s = %v, want no error", bindingName, err)
+	}
+
+	if binding.Spec.State != fleetv1beta1.BindingStateUnscheduled {
+		t.Errorf("binding state for binding %s = %v, want %v", bindingName, binding.Spec.State, fleetv1beta1.BindingStateUnscheduled)
```

Comment:
compare the whole struct to catch the regression if other fields are modified accidently?